src/components/AddComment.jsx

```diff
@@ -0,0 +1,85 @@
+import FloatingLabel from 'react-bootstrap/FloatingLabel';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import { postComment } from './utils';
+import { useState } from 'react';
+import { UserContext } from './UserContext';
+import { useContext } from 'react';
+
+export const AddComment = ({ setComments, id, setShowToast, setToastMsg }) => {
+	const { user } = useContext(UserContext);
+	const [newComment, setNewComment] = useState();
+	const [isLoading, setIsLoading] = useState();
+	const [btnMsg, setBtnMsg] = useState('Post');
+
+	const showError = (msg) => {
+		setToastMsg('Your comment was not posted! Pleas try again.');
+		setShowToast(true);
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		const commentBeingPosted = newComment;
+		if (commentBeingPosted === '') {
+			return;
+		}
+		const body = { username: user, body: newComment };
+		const addComment = {
+			body: newComment,
+			author: user,
+			article_id: id,
+			comment_id: crypto.randomUUID(),
+			created_at: Date(),
+			votes: 0,
+		};
+		setIsLoading(true);
+		setBtnMsg('Posting...');
```

Comment:
I would maybe change this to "Posted", can be a bit misleading if the user things it is still in the process of being posted (so they cant leave the site)